### ISSUE TITLE
How to replicate NER experiment that has only train and test sets?

### ISSUE THREAD

**Issue by Mark-L6n** [flairNLP/flair#1139]
I am trying to replicate a published NER experiment on a dataset that has only train and test sets.  How should I configure Flair to do this? 
I have tried:

``` python
corpus: Corpus = ColumnCorpus(data_folder, columns,
                              tag_to_bioes="ner", 
                              train_file='train.tsv',
                              test_file='test.tsv')
```
In training, there is: `train_with_dev: "False"` 
Also, there is:

```
EPOCH 1 done: loss 2.1065 - lr 0.1000
DEV : loss 0.7861679196357727 - score 0.8984
```
Since there is 1) a `DEV` score appearing, even though there is no dev set, 2) no `TEST` score, and 3) a suspiciously-excellent final result, I was wondering if the `test_file` was being used somehow in training with this setup.
If the `test_file` was in fact being utilized in training, is there a way to prevent this, so I can replicate previously-reported experiments? 

Experimenting, I tried specifying `corpus._dev = []`, but that resulted in a `ZeroDivisionError` following 1st epoch when calculating loss, apparently: `eval_loss /= batch_no`  in `models/sequence_tagger_model.py`.

**Comment by alanakbik** [comment]
Hello @Mark-L6n by default, the data loaders automatically sample a dev set from the training data if no dev set is provided. This is default behavior since the dev set is used for the annealing strategy and early stopping during training. 

If you do not want a dev set to be sampled, you need to set `train_with_dev=True` when you start the training run. In this case, the (sampled) dev set is integrated back into the train set for training. The `ModelTrainer` then switches to a different modus in which annealing is not done against dev score, but instead train loss. Training termines once the learning rate becomes too small or `max_epochs` is reached.

**Comment by Mark-L6n** [comment]
Thank you for the explanation @alanakbik!
So if I'm understanding this correctly, either setting of `train_with_dev` would be OK for comparing results with previous experiments, since nothing comes out of the test set either way to influence the training process, correct?

**Comment by alanakbik** [comment]
Yes, the test set never influences the training process.
